### PR TITLE
Migration guide

### PR DESCRIPTION
**Pull request by aszecowka**
This PR provide detailed instruction how to migrate Jenkins job to Prow job.

### CODE REVIEW

File: docs/prow/migration-guide.MD [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+ 
+
+#### Review and merge your PR. 
+After merging, job configuration is automatically applied on a Prow production cluster 
+and comment is added to the PR:
+![msg](./assets/msg-updated-config.png)
+
+There are following options to trigger your job:
+- create a new PR
+- request executing all tests for already created PR by adding comment: `/test all`
```
Comment (kazydek) [inline docs/prow/migration-guide.MD]: How to run a separate test?

File: docs/prow/migration-guide.MD [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+### Create presubmits and postsbumit Prow jobs.
+Presubmit job runs on a Pull Request (PR). Postsubmit job runs on merging to a branch.
+All jobs needs to be stored in `prow/jobs` directory. 
+Structure of this directory is following:
+```
+jobs
+|-- <repository1>
+|---- components
+|------ <component1>
+|-------- <component1.yaml>
```
[the remark this answers is not in the document]
Comment (aszecowka) [reply docs/prow/migration-guide.MD]: we decided to remove "jobs" from file name

File: docs/prow/migration-guide.MD [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+In your job, you should call a `pipeline.sh` script from provided buildpack image. This script executes makefile's rule defined for your component.
+Script `pipeline.sh` requires the environment variable `SORUCES_DIR` which is a path to a directory where a `Makefile` for your component is defined. 
+
+The content of the job file should be the following:
+```yaml
+presubmits: # runs on PRs
+  kyma-project/kyma:
+    - name: prow/kyma/components/binding-usage-controller
+      run_if_changed: "components/binding-usage-controller/"
```
Comment (michal-hudy) [inline docs/prow/migration-guide.MD]:
```suggestion
      run_if_changed: "^components/binding-usage-controller/"
```

File: docs/prow/migration-guide.MD [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+          base_ref: master
+          path_alias: github.com/kyma-project/test-infra
+      labels:
+        preset-dind-enabled: "true"
+        preset-sa-gcr-push: "true"
+        preset-docker-push-repository: "true"
+        preset-build-pr: "true"
+      spec:
+        containers:
+          - image: eu.gcr.io/kyma-project/prow/buildpack-golang:0.0.1
```
Comment (michal-hudy) [inline docs/prow/migration-guide.MD]: Use new image

File: docs/prow/migration-guide.MD [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+### Review and merge your PR on test-infra
+After merging, job configuration is automatically applied on a Prow production cluster 
+and comment is added to the PR:
+![msg](./assets/msg-updated-config.png).
+This is done by `config_updater` configured in `prow/plugins.yaml` file.
+
+ 
+### Create or adjust the `Makefile` for your component.
+Both of the buildpacks require a `Makefile` defined in your component directory.   
+The `Makefile` has to define 3 rules:
```
Comment (michal-hudy) [inline docs/prow/migration-guide.MD]:
```suggestion
The `Makefile` has to define 3 targets:
```

File: docs/prow/migration-guide.MD [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    repo: test-infra
+    base_ref: master
+    path_alias: github.com/kyma-project/test-infra
+  spec:
+    containers:
+    - image: eu.gcr.io/kyma-project/prow/buildpack-golang:0.0.1 
+      securityContext:
+        privileged: true
+      command:
+
```
Comment (michal-hudy) [inline docs/prow/migration-guide.MD]: Remove this new line

File: docs/prow/migration-guide.MD [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+  preset-sa-gcr-push: "true"
+  preset-docker-push-repository: "true"
+
+presubmits: # runs on PRs
+  kyma-project/kyma:
+  - name: &name prow/kyma/components/binding-usage-controller
+    context: *name 
+    branches:
+    - master
+    run_if_changed: "components/binding-usage-controller/"
```
Comment (michal-hudy) [inline docs/prow/migration-guide.MD]:
```suggestion
    run_if_changed: "^components/binding-usage-controller/"  
```

File: docs/prow/migration-guide.MD [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+        - master
+      <<: *job_template
+      labels:
+        <<: *job_labels_template
+        preset-build-master: "true"
+
+```
+
+To check if your configuration is correct, write a Go test. See the `development/tools/jobs/binding_usage_controller_test.go` file for reference.
+Your new test should be placed in `development/tools/jobs` to be executed by the presubmit job `prow/test-infra/test-jobs-yaml-definitions`.
```
Comment (kazydek) [inline docs/prow/migration-guide.MD]:
```suggestion
Place your new test under `development/tools/jobs` for the `prow/test-infra/test-jobs-yaml-definitions` presubmit job to execute it.
```

File: docs/prow/migration-guide.MD [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+In case you have access to the Prow cluster, there is an option to test prowjob on it. For a details, see [official documentation](https://github.com/kubernetes/test-infra/blob/master/prow/build_test_update.md#how-to-test-a-prowjob)
```
Comment (kazydek) [inline docs/prow/migration-guide.MD]:
```suggestion
If you have access to the Prow cluster, there is an option to test a Prow job on it. For details, see the [official documentation](https://github.com/kubernetes/test-infra/blob/master/prow/build_test_update.md#how-to-test-a-prowjob)
```

**Pull request merged**